src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -32,49 +33,114 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
+                {
+                    result.Add(new AddConstructorParametersCodeAction(
+                        document,
+                        constructorCandidate,
+                        containingType,
+                        constructorCandidate.MissingParameters,
+                        useSubMenuName: false));
+                }
+                result.Add(GetOptionalContructorParametersCodeAction(
+                    document,
+                    constructorCandidate,
+                    containingType,
+                    useSubMenuName: false));
             }
+            else
+            {
+                // Create sub menus for suggested actions, one for required parameters and one for optional parameters
+                var requiredParameterCodeActions = ArrayBuilder<CodeAction>.GetInstance();
+                var optionalParameterCodeActions = ArrayBuilder<CodeAction>.GetInstance();
+                foreach (var constructorCandidate in state.ConstructorCandidates)
+                {
+                    if (CanHaveRequiredParameters(constructorCandidate.Constructor.Parameters))
+                    {
+                        requiredParameterCodeActions.Add(new AddConstructorParametersCodeAction(
+                            document,
+                            constructorCandidate,
+                            containingType,
+                            constructorCandidate.MissingParameters,
+                            useSubMenuName: true));
+                    }
+                    optionalParameterCodeActions.Add(GetOptionalContructorParametersCodeAction(
+                        document,
+                        constructorCandidate,
+                        containingType,
+                        useSubMenuName: true));
+                }
 
-            var missingParameters = state.MissingParameters.SelectAsArray(p => CodeGenerationSymbolFactory.CreateParameterSymbol(
-                attributes: default,
-                refKind: p.RefKind,
-                isParams: p.IsParams,
-                type: p.Type,
-                name: p.Name,
-                isOptional: true,
-                hasDefaultValue: true));
+                if (requiredParameterCodeActions.Count > 0)
+                {
+                    result.Add(new CodeAction.CodeActionWithNestedActions(
+                        FeaturesResources.Add_parameter_to_constructor,
+                        requiredParameterCodeActions.ToImmutableAndFree(),
+                        isInlinable: false));
+                }
+
+                result.Add(new CodeAction.CodeActionWithNestedActions(
+                    FeaturesResources.Add_optional_parameter_to_constructor,
+                    optionalParameterCodeActions.ToImmutableAndFree(),
+                    isInlinable: false));
+            }
+
+            return result.AsImmutableOrNull();
```

Comment:
💡 I'm not a fan of intentionally returning `default(ImmutableArray<T>)`. It's a weird value type that likes to misbehave. Fortunately, it's not possible for this method to reach this case (all paths produce at least one action), so we can simplify this down to the following:

```suggestion
            return result.ToImmutableAndFree();
```

❕ The suggestion also corrects a failure to return the `ArrayBuilder<T>` to the pool.